From 78bec82ac03a36ba0649d8cd8bf90316cd254974 Mon Sep 17 00:00:00 2001
From: Bo Huang <bo.b.huang@intel.com>
Date: Wed, 31 Jul 2013 16:47:13 +0800
Subject: [PORT FROM MAIN]adb: optimize adb root procedure.

BZ: 117132

For adb root command, adbd should send message to remote firstly and
then restart itself. Otherwise, command can't be sent correctly.

Change-Id: Ib26708f128169431be6318e3bd6992d46fedb840
Orig-Change-Id: I1591b790608a359389a05bf80a0fa21253cb9ae0
Category: aosp improvement
Domain: USB ADB
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Bo Huang <bo.b.huang@intel.com>
---
 adb/services.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/adb/services.c b/adb/services.c
index 40ea473..a478303 100644
--- a/adb/services.c
+++ b/adb/services.c
@@ -75,9 +75,10 @@ void restart_root_service(int fd, void *cookie)
             return;
         }
 
-        property_set("service.adb.root", "1");
         snprintf(buf, sizeof(buf), "restarting adbd as root\n");
         writex(fd, buf, strlen(buf));
+        sleep(1);
+        property_set("service.adb.root", "1");
         adb_close(fd);
     }
 }
-- 
1.7.4.1

